### PR TITLE
Added Skiplist header file

### PR DESCRIPTION
**Pull request by PrakharUniyal**
Added a skiplist header file with the following methods:
- Insert an Element
- Create Levels in skiplist
- Delete an Element
- Search for an Element

Solves Issue #19

### CODE REVIEW

File: Stuff/skiplist.h [hunk 1]
```diff
@@ -0,0 +1,242 @@
+#pragma once
+
+#include <iostream>
+using namespace std;
+
+// Class to implement Node 
+class Node
```
Comment (Asmozan) [inline Stuff/skiplist.h]: Issue said it have to be template

Comment (Asmozan) [inline Stuff/skiplist.h]: issue said that it have to be template

File: Stuff/skiplist.h [hunk 2]
```diff
@@ -0,0 +1,242 @@
+#pragma once
+
+#include <iostream>
+using namespace std;
```
Comment (Asmozan) [inline Stuff/skiplist.h]: please, don't use "using namespace std" especially in header files

File: Stuff/skiplist.h [hunk 3]
```diff
@@ -0,0 +1,242 @@
+#pragma once
+
+#include <iostream>
+using namespace std;
+
+// Class to implement Node
```
Comment (Asmozan) [inline Stuff/skiplist.h]: there is no need for that comment, it reads directly from code

Comment (Asmozan) [reply Stuff/skiplist.h]: also remove forward no-needed comments

File: Stuff/skiplist.h [hunk 4]
```diff
@@ -0,0 +1,242 @@
...
+  
+    /* reached level 0 and advance pointer to  
+       right, which is possibly our desired node*/
+    current = current->forward[0]; 
+  
+    // If current node have key equal to 
+    // search key, we have found our target node 
+    if(current and current->key == key)  
+        cout<<"Found key: "<<key<<"\n"; 
+};
```
Comment (Asmozan) [inline Stuff/skiplist.h]: add new line at end of file

File: Stuff/skiplist.h [hunk 5]
```diff
@@ -0,0 +1,242 @@
...
+  
+    } 
+  
+    /* reached level 0 and advance pointer to  
+       right, which is possibly our desired node*/
+    current = current->forward[0]; 
+  
+    // If current node have key equal to 
+    // search key, we have found our target node 
+    if(current and current->key == key)
```
Comment (Asmozan) [inline Stuff/skiplist.h]: "and" is not "common sense" please change it to &&

File: Stuff/skiplist.h [hunk 6]
```diff
@@ -0,0 +1,242 @@
...
+                break; 
+  
+            update[i]->forward[i] = current->forward[i]; 
+        } 
+  
+        // Remove levels having no elements  
+        while(level>0 && 
+              header->forward[level] == 0) 
+            level--; 
+         cout<<"Successfully deleted key "<<key<<"\n";
```
Comment (Asmozan) [inline Stuff/skiplist.h]: keep couts like this only for debugin purposes, there is no need to see it while using it in production code, remove it or introduce debug flag

File: Stuff/skiplist.h [hunk 7]
```diff
@@ -0,0 +1,242 @@
...
+    current = current->forward[0];
+  
+    /* if current is NULL that means we have reached 
+       to end of the level or current's key is not equal 
+       to key to insert that means we have to insert 
+       node between update[0] and current node */
+    if (current == NULL || current->key != key)
+    { 
+        // Generate a random level for node 
+        int rlevel = randomLevel();
```
Comment (Asmozan) [inline Stuff/skiplist.h]: please change it to radnomLevel, as rlevel doesn't say much, it is r value level? rightLevel? I can see from declaration but it should be obvious without it

File: Stuff/skiplist.h [hunk 8]
```diff
@@ -0,0 +1,242 @@
...
+    /*    start from highest level of skip list 
+        move the current pointer forward while key  
+        is greater than key of node next to current 
+        Otherwise inserted current in update and  
+        move one level down and continue search 
+    */
+    for (int i = level; i >= 0; i--) 
+    { 
+        while (current->forward[i] != NULL && 
+              current->forward[i]->key < key)
```
Comment (Asmozan) [inline Stuff/skiplist.h]: please add braces after while, to make code more bug prune

File: Stuff/skiplist.h [hunk 9]
```diff
@@ -0,0 +1,242 @@
...
+        Node(int, int); 
+};
+
+//Node constructor  
+Node::Node(int key, int level) 
+{ 
+    this->key = key; 
+  
+    // Allocate memory to forward  
+    forward = new Node*[level+1];
```
Comment (Asmozan) [inline Stuff/skiplist.h]: looks like this memory is never freed, but correct me if I'm wrong

**Comment by PrakharUniyal** [comment]
I've worked on the issues you pointed out. Forgive me for the bad code this is the first time I'm making a contribution using C++.

**Comment by ze413X** [comment]
Im currently abroad. Will review in the following week.

**Comment by Asmozan** [comment]
PrakharUniyal nice, but it is still no template which is one of main things specified in Issue, also that I see
Must come with test code to not only show off the tree but also test it further. is not done either

**Comment by PrakharUniyal** [comment]
@Asmozan I need to do a bit of research on that. I'll try to correct it by tomorrow.

**Comment by ze413X** [comment]
Thank you for your submission.

**Comment by Asmozan** [comment]
hello, it is Your repo so a question out of curious, but why You merged it when your requirements were:
Must be a single header file.
Must be Template to allow storing different values.
Must come with test code to not only show off the tree but also test it further.

so only one of requirements were achieved? :V

**Comment by ze413X** [comment]
Honestly, I was dicking around in SourceTree which fucking awful dealing with pull requests. Since it was so low activity on the task I merged it (to review it in VS lol) and thought it would be much greater quality. Now I am just regretting it. It might be dismissed later, but atleast he got his hacktober points albeit the poor attempt.

**Pull request merged**